### PR TITLE
Add ability to create cloud creds for any node driver 

### PR DESCRIPTION
**Pull request by thedadams**
A previous PR (31548) added support for Google cloud credential.
However, this made the unsupported Google node driver active. In order
to deactivate the node driver and create the cloud credential schema for
the Google driver, a new field is added to the NodeDriverSpec:
AddCloudCredentialByDefault. If this is `true` for any node driver, then
the driver is downloaded, if necessary, and the cloud credential schema
created. This would work for any node driver, not just the built-ins.

Issue:
https://github.com/rancher/rancher/issues/31858

### CODE REVIEW

**Review by cmurphy — APPROVED** [review]
lgtm, tested it and it seems to fix the problem

**Review by mjura — DISMISSED** [review]
LGTM

**Review by kinarashah — APPROVED** [review]
Hm,

File: pkg/apis/management.cattle.io/v3/machine_types.go [hunk 1]
```diff
@@ -310,15 +310,16 @@ type Condition struct {
...
-	Checksum         string   `json:"checksum"`
-	UIURL            string   `json:"uiUrl"`
-	WhitelistDomains []string `json:"whitelistDomains,omitempty"`
+	DisplayName                 string   `json:"displayName"`
+	Description                 string   `json:"description"`
+	URL                         string   `json:"url" norman:"required"`
+	ExternalID                  string   `json:"externalId"`
+	Builtin                     bool     `json:"builtin"`
+	Active                      bool     `json:"active"`
+	AddCloudCredentialByDefault bool     `json:"addCloudCredentialByDefault"`
```
Comment (dramich) [inline pkg/apis/management.cattle.io/v3/machine_types.go:319]: Let's drop the `ByDefault` part and then add a comment above to explain what it does.

**Review by dramich — APPROVED** [review]

**Review by cbron — APPROVED** [review]
Dan/Kinara already approved, re-stamping.

**Review by cbron — APPROVED** [review]

**Pull request merged**